shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -382,9 +387,42 @@ public void destroy() {
         mNativeView = null;
     }
 
+    @Override
+    public void updateInputTarget(InputTarget target, boolean setAsTarget) {
+        InputTarget oldTarget;
+        if (mInputTarget == target) {
+            if (setAsTarget) {
+                // It's already the input target so there's nothing to do.
+                return;
+            }
+
+            // We'll remove it as the current target, and revert to the plugin as
+            // the new target. We'll also hide the keyboard.
+            oldTarget = target;
+            mInputTarget = mTextInputPlugin;
+        } else if (setAsTarget) {
+            // We'll set it as the current target.
+            oldTarget = mInputTarget;
+            mInputTarget = target;
+        } else {
+            // It's not the current target and we're not setting it as the new target.
+            return;
+        }
+
+        oldTarget.disposeInputConnection();
+        mImm.restartInput(mInputTarget.getTargetView());
+    }
+
     @Override
     public InputConnection onCreateInputConnection(EditorInfo outAttrs) {
-        return mTextInputPlugin.createInputConnection(this, outAttrs);
+        return mInputTarget.createInputConnection(outAttrs);
+    }
+
+    @Override
+    public boolean checkInputConnectionProxy(View view) {
+        // If the input target is not us, then it's a platform view so we're
```

Comment:
Would it be sufficient if I did something like:
```java
final View proxyTarget = mInputTarget.getTargetView();
return proxyTarget == view && proxyTarget != this;
```
This would only return true if the view that's requesting input is an InputTarget and is an embedded view i.e not the FlutterView.

Reply:
I believe it's possible for a platform view to have a view hierarchy where a subview that is not focused is interacting with the IMM (I believe that's what happening on webviews prior to Android N)

Reply:
Ah I see, so that's why we have to walk the hierarchy.

Also I tested it on my S4, which runs Lollipop, and it indeed doesn't work. It gains focus, and then loses it shortly after. I'll see if this helps.